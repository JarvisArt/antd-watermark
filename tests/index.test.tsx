import React from 'react';
import Watermark from '../src';
import { render, waitFor } from '@testing-library/react';

describe('Watermark', () => {
  const mockSrcSet = jest.spyOn(Image.prototype, 'src', 'set');

  beforeAll(() => {
    mockSrcSet.mockImplementation(function fn() {
      this.onload?.();
    });
  });

  afterAll(() => {
    mockSrcSet.mockRestore();
  });

  it('The watermark should render successfully', () => {
    const { container } = render(
      <Watermark className="watermark" content="Ant Design" />,
    );
    expect(container.querySelector('.watermark div')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it('The offset should be correct', () => {
    const { container } = render(
      <Watermark
        className="watermark"
        offset={[200, 200]}
        content={['Ant Design', 'Ant Design Pro']}
      />,
    );
    const target = container.querySelector<HTMLDivElement>('.watermark div');
    expect(target?.style.left).toBe('150px');
    expect(target?.style.top).toBe('150px');
    expect(target?.style.width).toBe('calc(100% - 150px)');
    expect(target?.style.height).toBe('calc(100% - 150px)');
    expect(container).toMatchSnapshot();
  });

  it('Interleaved watermark backgroundSize is correct', () => {
    const { container } = render(
      <Watermark
        className="watermark"
        width={200}
        height={200}
        content="Ant Design"
        gap={[100, 100]}
      />,
    );
    const target = container.querySelector<HTMLDivElement>('.watermark div');
    expect(target?.style.backgroundSize).toBe('600px');
    expect(container).toMatchSnapshot();
  });

  it('Image watermark snapshot', () => {
    const { container } = render(
      <Watermark image="https://gw.alipayobjects.com/zos/bmw-prod/59a18171-ae17-4fc5-93a0-2645f64a3aca.svg" />,
    );
    expect(container).toMatchSnapshot();
  });

  it('MutationObserver should work properly', async () => {
    const { container } = render(
      <Watermark className="watermark" content="MutationObserver" />,
    );
    const target = container.querySelector<HTMLDivElement>('.watermark div');
    target?.remove();
    await waitFor(() => expect(target).toBeTruthy());
    expect(container).toMatchSnapshot();
  });

  it('Observe the modification of style', async () => {
    const { container } = render(
      <Watermark
        offset={[-200, -200]}
        className="watermark"
        content="MutationObserver"
      />,
    );
    const target = container.querySelector<HTMLDivElement>('.watermark div');
    target?.setAttribute('style', '');
    await waitFor(() => expect(target).toBeTruthy());
    expect(container).toMatchSnapshot();
  });
});
