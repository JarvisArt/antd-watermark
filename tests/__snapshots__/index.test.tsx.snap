// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Watermark Image watermark snapshot 1`] = `
<div>
  <div
    style="position: relative;"
  >
    <div
      style="z-index: 9; position: absolute; left: 0; top: 0; width: 100%; height: 100%; pointer-events: none; background-repeat: repeat; background-position: 0px 0px; background-image: url('data:image/png;base64,00'); background-size: 440px;"
    />
  </div>
</div>
`;

exports[`Watermark Interleaved watermark backgroundSize is correct 1`] = `
<div>
  <div
    class="watermark"
    style="position: relative;"
  >
    <div
      style="z-index: 9; position: absolute; left: 0; top: 0; width: 100%; height: 100%; pointer-events: none; background-repeat: repeat; background-position: 0px 0px; background-image: url('data:image/png;base64,00'); background-size: 600px;"
    />
  </div>
</div>
`;

exports[`Watermark MutationObserver should work properly 1`] = `
<div>
  <div
    class="watermark"
    style="position: relative;"
  >
    <div
      style="z-index: 9; position: absolute; left: 0; top: 0; width: 100%; height: 100%; pointer-events: none; background-repeat: repeat; background-position: 0px 0px; background-image: url('data:image/png;base64,00'); background-size: 232px;"
    />
  </div>
</div>
`;

exports[`Watermark Observe the modification of style 1`] = `
<div>
  <div
    class="watermark"
    style="position: relative;"
  >
    <div
      style="z-index: 9; position: absolute; left: 0; top: 0; width: 100%; height: 100%; pointer-events: none; background-repeat: repeat; background-position: -250px -250px; background-image: url('data:image/png;base64,00'); background-size: 232px;"
    />
  </div>
</div>
`;

exports[`Watermark The offset should be correct 1`] = `
<div>
  <div
    class="watermark"
    style="position: relative;"
  >
    <div
      style="z-index: 9; position: absolute; left: 150px; top: 150px; width: calc(100% - 150px); height: calc(100% - 150px); pointer-events: none; background-repeat: repeat; background-position: 0px 0px; background-image: url('data:image/png;base64,00'); background-size: 228px;"
    />
  </div>
</div>
`;

exports[`Watermark The watermark should render successfully 1`] = `
<div>
  <div
    class="watermark"
    style="position: relative;"
  >
    <div
      style="z-index: 9; position: absolute; left: 0; top: 0; width: 100%; height: 100%; pointer-events: none; background-repeat: repeat; background-position: 0px 0px; background-image: url('data:image/png;base64,00'); background-size: 220px;"
    />
  </div>
</div>
`;
